cypress/e2e/signIn.cy.js

```diff
@@ -10,29 +10,37 @@ const homePage = new HomePageObject();
 describe('Sign In page', () => {
   let user;
 
-  before(() => {
+  beforeEach(() => {
     cy.task('db:clear');
     cy.task('generateUser').then(generateUser => {
       user = generateUser;
     });
+    signInPage.visit();
   });
 
   it('should provide an ability to log in with existing credentials', () => {
-    signInPage.visit();
-    cy.register(user.email, user.username, user.password);
-
-    signInPage.emailField
-      .type(user.email);
-    signInPage.passwordField
-      .type(user.password);
-    signInPage.signInBtn
-      .click();
-
+    cy.registerNewUser().then(user => {
+      cy.getByDataCy('email-sign-in').type(user.email);
```

Comment:
The test 'should provide an ability to log in with existing credentials' does not actually show a successful login. It only checks if the username link exists after clicking the login button. It would be better to check if the login was successful by checking if the user is redirected to the expected page.